File: client/src/app/_models/incident.ts
import { Call } from "./call";
import { Crew } from "./crew";
import { Device } from "./device";
import { Resolution } from "./resolution";

export interface Incident {
    id?: number;
    type: string;
    priority?: string;
    isConfirmed: string;
    status?: string;
    location: string;
    latitude?: number;
    longitude?: number;
    estimatedTimeOfTheCrewArrival?: Date;
    actualTimeOfTheCrewArrival?: Date;
    outageTime: Date;
    estimatedTimetoRestore?: Date;
    affectedCustomers?: number;
    numberOfCalls?: number;
    voltage?: number;
    scheduledTime?: Date;
    resolutionId?: number;
    crewId?: number;
}

File: backend/Entities/Incident.cs
using System;
using System.Collections.Generic;

namespace backend.Entities
{
    public class Incident
    {
        public int Id { get; set; }
        public string Type { get; set; }
        public int? Priority { get; set; }
        public bool isConfirmed { get; set; } 
        public string Status { get; set; }
        public string Location { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public DateTime? EstimatedTimeOfTheCrewArrival { get; set; }
        public DateTime? ActualTimeOfTheCrewArrival { get; set; }
        public DateTime? OutageTime { get; set; }
        public DateTime? EstimatedTimetoRestore { get; set; }
        public int? AffectedCustomers { get; set; }
        public int? NumberOfCalls { get; set; }
        public double? Voltage { get; set; }
        public DateTime? ScheduledTime { get; set; }

        public int? TakenToResolveUserId { get; set; } 
        public User User { get; set; }

        public int? ResolutionId { get; set; }
        public Resolution Resolution { get; set; }

        public int? CrewId { get; set; }
        public Crew Crew { get; set; }
        
        public ICollection<Device> Devices { get; set; }
        public ICollection<Call> Calls { get; set; }
        public ICollection<IncidentPhoto> Photos { get; set; }
        public ICollection<WorkRequest> WorkRequests { get; set; }
        public ICollection<WorkPlan> WorkPlans { get; set; }

    }
}